tree/tree.go

```diff
@@ -118,7 +151,64 @@ func (t *Tree) String() string {
 	return t.ensureRenderer().render(t, true, "")
 }
 
-// Child adds a child to this tree.
+// SetChildren overwrites a Tree's Children.
+func (t *Tree) SetChildren(children ...any) *Tree {
+	t.children = NodeChildren(nil)
+	return t.Child(children)
+}
+
+// Replace swaps the child at the given index with the given child.
+func (t *Tree) Replace(index int, child any) *Tree {
+	nodes := t.anyToNode(child)
+	t.children = t.children.(NodeChildren).Replace(index, nodes[0])
+	return t
+}
+
+// Insert child at the given index.
+func (t *Tree) Insert(index int, child any) *Tree {
+	nodes := t.anyToNode(child)
+	t.children = t.children.(NodeChildren).Insert(index, nodes[0])
+	return t
+}
+
+// TODO probably don't need this to be an []any
+func (t *Tree) anyToNode(children ...any) []Node {
+	var nodes []Node
+	for _, child := range children {
+		switch item := child.(type) {
+		case *Tree:
+			child, _ := child.(*Tree)
+			nodes = append(nodes, child)
+		case Children:
+			for i := 0; i < item.Length(); i++ {
+				nodes = append(nodes, item.At(i))
+			}
+		case Node:
+			nodes = append(nodes, item)
+		case fmt.Stringer:
```

Comment:
Might be useful to support `encoding.TextMarshaler` as well

Reply:
We don't have this in other type checks in `tree`, maybe something we can add down the line